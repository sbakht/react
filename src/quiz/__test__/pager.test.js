import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Pager from '../pager';
import toJSON from 'enzyme-to-json';

test('render first page', () => {
  var fn = jest.fn();
  var component = shallow(<Pager qNum={1} qLen={2} onPrev={fn} onNext={fn} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('render last page', () => {
  var fn = jest.fn();
  var component = shallow(<Pager qNum={2} qLen={2} onPrev={fn} onNext={fn} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('render middle page', () => {
  var fn = jest.fn();
  var component = shallow(<Pager qNum={2} qLen={3} onPrev={fn} onNext={fn} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('on click prev page', () => {
  var fn = jest.fn();
  var prev = jest.fn();
  var component = shallow(<Pager qNum={2} qLen={3} onPrev={prev} onNext={fn} />);
  component.find("li a").at(0).simulate('click');
  expect(prev.mock.calls.length).toBe(1);
});

test('on click next page', () => {
  var fn = jest.fn();
  var next = jest.fn();
  var component = shallow(<Pager qNum={2} qLen={3} onPrev={fn} onNext={next} />);
  component.find("li a").at(1).simulate('click');
  expect(next.mock.calls.length).toBe(1);
});

test('has proptypes', () => {
  expect(Pager.propTypes).toBeDefined();
  
  Object.keys(Pager.propTypes).map((key) => {
    expect(Pager.propTypes[key]).toBeDefined();
  });
});