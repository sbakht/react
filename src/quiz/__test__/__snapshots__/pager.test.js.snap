// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render first page 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Next
        </a>
      </li>
    </ul>
  </nav>
  <div />
</div>
`;

exports[`render last page 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Previous
        </a>
      </li>
    </ul>
  </nav>
  <div>
    <button
      type="button"
    >
      Results
    </button>
  </div>
</div>
`;

exports[`render middle page 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Previous
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          Next
        </a>
      </li>
    </ul>
  </nav>
  <div />
</div>
`;
